code-challenges.test.js

```diff
@@ -13,34 +15,69 @@
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
-
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe('describeArray', () => {
+  it('Takes in an array of objects and returns an array with a sentence about each person with their name capitalized.', () => {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(describeArray(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
 
 // b) Create the function that makes the test pass.
+// To create this function, I will first need to  take in the given array as an argument. Then, I will map over that array, where each element will be an object. I will take the value associated with the name key from the current element, and split it into an array of strings, based on where a space is. Then, with this new array, I will want to capitalize each value, so I will use another map to capitalize each word and then join them back into one string. From there, I only need to add on the rest of the sentance using string interpolation. I will then return this entire statement.
+const describeArray = (array) => {
+  return array.map((element) => element.name.split(" ").map((value) => value = (value[0].toUpperCase()) + value.slice(1)).join(" ") + ` is ${element.occupation}.`)
+}
```

Comment:
Nice logic! The pseudo code could be made more concise, but I like that you touched every aspect of the code's functionality